corehq/apps/domain/tests/test_forms.py

```diff
@@ -29,119 +21,239 @@ def test_visible_fields(self):
+
+        form = self._create_form(operator_call_limit=95)
+        form.save(Mock(), self.mock_domain)
+
+        self.assertEqual(call_settings.call_limit, 95)
+        call_settings.save.assert_called()
+
+# Helpers
+    def setUp(self):
+        super().setUp()
+        self.mock_domain = self._create_mock_domain()
+        self.ucr_toggle_enabled = True
+        self.sms_user_provisioning_toggle_enabled = True
 
-    def create_form(self, domain=None, **kwargs):
+        mock_ucr_toggle = patch.object(forms.MOBILE_UCR, 'enabled',
+            side_effect=lambda domain: self.ucr_toggle_enabled)
+        mock_ucr_toggle.start()
+        self.addCleanup(mock_ucr_toggle.stop)
+
+        mock_sms_provisioning_toggle = patch.object(forms.TWO_STAGE_USER_PROVISIONING_BY_SMS, 'enabled',
+            side_effect=lambda domain: self.sms_user_provisioning_toggle_enabled)
+        mock_sms_provisioning_toggle.start()
+        self.addCleanup(mock_sms_provisioning_toggle.stop)
+
+        mock_get_domain_by_name = patch.object(Domain, 'get_by_name', return_value=self.mock_domain)
+        mock_get_domain_by_name.start()
+        self.addCleanup(mock_get_domain_by_name.stop)
+
+        mock_call_limit_domain_patcher = patch.object(OperatorCallLimitSettings,
+            'objects')
+        self.mock_call_limit_settings = mock_call_limit_domain_patcher.start()
+        self.mock_call_limit_settings.values_list.return_value = []
+        self.addCleanup(mock_call_limit_domain_patcher.stop)
+
+    def _create_form(self, can_use_custom_logo=True, call_center_enabled=True, **kwargs):
+        self.mock_domain.call_center_config.enabled = call_center_enabled
         data = {
-            "hr_name": "foo",
-            "project_description": "sample",
-            "default_timezone": "UTC",
+            'hr_name': 'foo',
+            'project_description': 'sample',
+            'default_timezone': 'UTC',
+            'confirmation_link_expiry': 500
         }
-        if kwargs:
-            for field, value in kwargs.items():
-                data.update({field: value})
-        if not domain:
-            domain = self.domain
-        return DomainGlobalSettingsForm(data, domain=domain)
-
-    def tearDown(self):
-        set_toggle(TWO_STAGE_USER_PROVISIONING_BY_SMS.slug, self.domain, False, namespace=NAMESPACE_DOMAIN)
-        self.domain.delete()
-        OperatorCallLimitSettings.objects.all().delete()
-        super().tearDown()
+        data.update(**kwargs)
+        return DomainGlobalSettingsForm(
+            data, domain=self.mock_domain, can_use_custom_logo=can_use_custom_logo)
+
+    def _create_mock_domain(self, name='test-domain', **kwargs):
+        domain_properties = {
+            'confirmation_link_expiry_time': 500,
+            'call_center_enabled': False,
+            'default_timezone': 'UTC'
+        }
+        domain_properties.update(kwargs)
+        domain_obj = Mock(**domain_properties)
```

Comment:
Is there a reason why this can't use a `corehq.apps.domain.models.Domain` object here instead of `Mock`? (The `Domain` object does not need to be saved.)

The same question applies in other tests as well.

Reply:
The `TestCase` version of the code had a mocked domain, and I think I probably just copied that. I agree that using the actual `Domain` should be possible, so I'll give that a try

Reply:
Addressed in 623a55057912490798a4e2002f4eedbd009a59cb